Add org.srb2.srb2-legacy

P-AS opened the pull request:
<!-- ⚠️⚠️ Submission pull request MUST be made against the `new-pr` **base branch** ⚠️⚠️  -->

<!-- 💡 Go to the preview tab to click the links below 💡 -->

### Please confirm your submission meets all the criteria

<!-- 💡 Please replace each `[ ]` with `[X]` when the step is complete 💡 -->

- [X] Please describe the application briefly. 
SRB2 Legacy is an updated fork of [Sonic Robo Blast 2](https://www.srb2.org/) version 2.1 that includes bugfixes and QOL improvements from future releases. This submission is an adaption of the manifest [from upstream SRB2](https://github.com/flathub/org.srb2.SRB2).
- [X] The domain used for the application ID is [controlled by the application developer(s)][appid-domain] and the [application id guidelines][appid] are followed.
The source code is hosted at https://git.do.srb2.org/PAS/srb2-legacy. Other forks of SRB2 that are hosted there also use the `org.srb2` components for flathub.
- [X] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor to the project. If not, I contacted upstream developers about this submission. **Link:** https://git.do.srb2.org/PAS/srb2-legacy

<!-- ⚠️⚠️ DO NOT modify anything below this line ⚠️⚠️  -->

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

bbhtt commented:
Why does this need a legacy version? Creating a submission for "legacy" every time they release a major update is not sustainable.

bbhtt commented on org.srb2.srb2-legacy.metainfo.xml:
```diff
@@ -0,0 +1,54 @@
+  <launchable type="desktop-id">org.srb2.srb2-legacy.desktop</launchable>
+  <screenshots>
+    <screenshot type="default">
+      <image>https://file.garden/ZuUiCPHqjV5ssvQM/metadata/tiltescreen.png</image>
```
If this is a temporary filehost, then don't use it.

I also can't reach https://file.garden

P-AS replied:
It is a permanent filehost. The images are uploaded to my account. I'm not sure why you are unable to reach it. I could use github as an alternative, but then the links would look like this: https://github.com/user-attachments/assets/8eb891b7-47d6-4d02-8daa-1f639a5ef870. Would it be better to switch it to github images?

P-AS replied:
I have decided to move them to https://git.do.srb2.org/PAS/srb2-legacy/-/wikis/Screenshots. That way you should be able to access them, and it's directly connected to the git.do.srb2.org repo.

bbhtt commented on org.srb2.srb2-legacy.metainfo.xml:
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
This file should be upstream.

bbhtt commented on org.srb2.srb2-legacy.yaml:
```diff
@@ -0,0 +1,97 @@
+      - --disable-tests
+      - --without-portaudio
+      - --without-portaudiocpp
+    cleanup:
```
All these cleanups should be in top level, no need to repeat it for every modile

hfiguiere commented on org.srb2.srb2-legacy.yaml:
```diff
@@ -0,0 +1,97 @@
+
+  - name: srb2legacy
+    buildsystem: simple
+    build-options:
+      arch:
+        aarch64:
+          env:
+            ARCH_MAKE_ARGS: LINUX64=1
+        x86_64:
+          env:
+            ARCH_MAKE_ARGS: LINUX64=1
```
```suggestion
    build-options:
      env:
        ARCH_MAKE_ARGS: LINUX64=1
```

hfiguiere replied:
or maybe just set it on line 65.

P-AS commented:
> Why does this need a legacy version? Creating a submission for "legacy" every time they release a major update is not sustainable.

The main reason for 2.1 to have one is that the content is almost completely different. All the music, levels, sprites were remade in 2.2. The content is distinct enough to warrant having a separate version. I don't know if that will still be true for future releases, though.

P-AS commented:
I have:
- Moved the metainfo file upstream
- Moved the cleanup to the top level
- Moved `LINUX64=1` directly to the `make` command
- Moved the screenshots off filegarden to https://git.do.srb2.org/PAS/srb2-legacy/-/wikis/Screenshots
I hope this is valid now.

bbhtt commented:
> The main reason for 2.1 to have one is that the content is almost completely different. All the music, levels, sprites were remade in 2.2. The content is distinct enough to warrant having a separate version. I don't know if that will still be true for future releases, though.

This seems a bit unsustainable if they do another set of major changes and someone wants to make a another Flatpak.

It would be valid if they broke game data or config file compatibility again under very specific circumstances.

bbhtt commented:
bot, build org.srb2.srb2-legacy
